app/src/components/data-grid/DataGridValidationAlert.tsx

```diff
@@ -98,8 +98,14 @@ const DataGridValidationAlert = <RowType extends Record<any, any>>(props: IDataG
       }
 
       const field = String(focusedError.field);
+      const isFieldVisible = props.muiDataGridApiRef.getVisibleColumns().some((column) => column.field === field);
+
+      if (!isFieldVisible) {
+        return;
+      }
+
       const rowIndex = props.muiDataGridApiRef.getSortedRowIds().indexOf(focusedError.rowId);
-      const colIndex = props.muiDataGridApiRef.getColumnIndex(field);
+      const colIndex = props.muiDataGridApiRef.getColumnIndexRelativeToVisibleColumns(field);
```

Comment:
Nice find. 

Also, yet another situation where if you search "getColumnIndexRelativeToVisibleColumns" you get exactly 1 search result (from the MUI doc). Crazy.

Reply:
Hmm, I'm still getting an error when I click through the banner errors:

```
visibleColumns[params.colIndex] is undefined
./node_modules/@mui/x-data-grid/hooks/features/scroll/useGridScroll.js/useGridScroll/scrollToIndexes<@http://localhost:7107/static/js/bundle.js:150323:9
./src/components/data-grid/DataGridValidationAlert.tsx/DataGridValidationAlert/focusErrorAtIndex<@http://localhost:7107/static/js/bundle.js:3164:29
./src/components/data-grid/DataGridValidationAlert.tsx/DataGridValidationAlert/handleNext</<@http://localhost:7107/static/js/bundle.js:3138:24
basicStateReducer@http://localhost:7107/static/js/bundle.js:266236:45
updateReducer@http://localhost:7107/static/js/bundle.js:266344:26
updateState@http://localhost:7107/static/js/bundle.js:266626:14
useState@http://localhost:7107/static/js/bundle.js:267327:20
useState@http://localhost:7107/static/js/bundle.js:287185:25
DataGridValidationAlert@http://localhost:7107/static/js/bundle.js:3100:76
```

Reply:
Also, not sure if its critical right this second, but the validation banner error count is only considering the visible columns. So even if 12 columns are wrong, if only 5 are visible, it just shows '1/5' in the banner. Which, maybe is fine for now?

Reply:
> Also, not sure if its critical right this second, but the validation banner error count is only considering the visible columns. So even if 12 columns are wrong, if only 5 are visible, it just shows '1/5' in the banner. Which, maybe is fine for now?

I'm not able to reproduce this behaviour 🤔